skimage/color/colorconv.py

```diff
@@ -132,54 +140,39 @@ def convert_colorspace(arr, fromspace, tospace):
     return todict[tospace](fromdict[fromspace](arr))
 
 
-def _prepare_colorarray(arr):
+def _prepare_colorarray(arr, force_copy=False):
     """Check the shape of the array and convert it to
     floating point representation.
     """
     arr = np.asanyarray(arr)
 
-    if arr.ndim not in [3, 4] or arr.shape[-1] != 3:
-        msg = ("the input array must be have a shape == (.., ..,[ ..,] 3)), " +
-               "got (" + (", ".join(map(str, arr.shape))) + ")")
-        raise ValueError(msg)
+    if arr.ndim < 2 or arr.shape[-1] != 3:
+        raise ValueError("Input array must have a shape == (.., 3)), "
+                         "got {0}".format(arr.shape))
 
-    return dtype.img_as_float(arr)
-
-
-def _prepare_rgba_array(arr):
-    """Check the shape of the array to be RGBA and convert it to
-    floating point representation.
-    """
-    arr = np.asanyarray(arr)
-
-    if arr.ndim not in [3, 4] or arr.shape[-1] != 4:
-        msg = ("the input array must have a shape == (.., ..,[ ..,] 4)), "
-               "got {0}".format(arr.shape))
-        raise ValueError(msg)
-
-    return dtype.img_as_float(arr)
+    return dtype.img_as_float(arr, force_copy=force_copy)
 
 
 def rgba2rgb(rgba, background=(1, 1, 1)):
-    """RGBA to RGB conversion.
+    """RGBA to RGB conversion using alpha blending [1]_.
 
     Parameters
     ----------
-    rgba : array_like
-        The image in RGBA format, in a 3-D array of shape ``(.., .., 4)``.
+    rgba : (.., 4) array_like
+        The image in RGBA format. Final dimension denotes channels.
     background : array_like
-        The color of the background to blend the image with. A tuple
-        containing 3 floats between 0 to 1 - the RGB value of the background.
+        The color of the background to blend the image with (3 floats
+        between 0 to 1 - the RGB value of the background).
 
     Returns
     -------
-    out : ndarray
-        The image in RGB format, in a 3-D array of shape ``(.., .., 3)``.
+    out : (.., 3) ndarray
+        The image in RGB format. Same dimensions as input.
 
     Raises
     ------
     ValueError
-        If `rgba` is not a 3-D array of shape ``(.., .., 4)``.
+        If `rgba` is not at least 2-D with shape (.., 4).
```

Comment:
Should we add a note to say that the output is float (and <1)? It was not obvious to me, like with a default background I could have expected that just the last channel was dropped (but ok for this one could also do just slicing). @rfezzani happy to open an issue for this instead because this has nothing to do with your PR.